xls-transaction-auth.md

```diff
@@ -0,0 +1,218 @@
+
+Currently, critical issuer actions, such as authorizing trustlines, require direct control by the account's keys, hindering operational efficiency and complex use cases. By empowering account holders to selectively delegate specific permissions to other accounts, this proposal aims to enhance account usability without compromising security. This mechanism will unlock new possibilities for XRPL applications, such as multi-party workflows and advanced account management strategies.
+
+## 1. Overview
+
+We propose:
+* Creating a `TransactionAuth` ledger object.
+* Creating a `TransactionAuth` transaction type.
+
+We also propose modifying the transaction common fields.
+
+This feature will require an amendment, tentatively titled `featureTransactionAuth`.
+
+### 1.1. Basic Flow
+
+Isaac, a token issuer, wants to set up his account to follow security best practices and separation of responsibilities. He wants Alice to manage token issuing and Bob to manage trustlines. He is also working with Kylie, a KYC provider, who he wants to be able to authorize trustlines but not have any other permissions (as she is an external party).
+
+He can authorize:
+* Alice's account for the `Payment` transaction permission.
+* Bob's account for the `TrustSet` transaction permission.
+* Kylie's account for the `TrustlineAuthorize` granular permission.
+
+## 2. On-Ledger Object: `TransactionAuth`
+
+This object represents a set of permissions that an account has delegated to another account, and is modeled to be similar to [`DepositPreauth` objects](https://xrpl.org/docs/references/protocol/ledger-data/ledger-entry-types/depositpreauth).
+
+### 2.1. Fields
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`LedgerIndex`| ✔️|`string`|`Hash256`|The unique ID of the ledger object.|
+|`LedgerEntryType`| ✔️|`string`|`UInt16`|The ledger object's type (`TransactionAuth`)|
+|`Account`| ✔️|`string`|`AccountID`|The account that wants to authorize another account.|
+|`Authorize`| ✔️|`string`|`AccountID`|The authorized account.|
+|`Permissions`| ✔️|`string`|`STArray`|The transaction permissions that the account has access to.|
+|`PreviousTxnID`|✔️|`string`|`Hash256`|The identifying hash of the transaction that most recently modified this object.|
+|`PreviousTxnLgrSeqNumber`|✔️|`number`|`UInt32`|The index of the ledger that contains the transaction that most recently modified this object.|
+
+#### 2.1.1. Object ID
+
+The ID of this object will be a hash of the `Account` and `Authorize` fields, combined with the unique space key for `TransactionAuth` objects, which will be defined during implementation.
+
+#### 2.1.2. `Permissions`
+
+This field is an array of permissions to delegate to the account, as listed in [XLS-otherd, Account Transaction Permissions](https://gist.github.com/mvadari/a8d76f0c4e3aa54eb765f08bcacc5316). The array will have a maximum length of 10.
```

Comment:
what if the account want to delegate more than ten permissions?

Reply:
Then you'll need to create another account to delegate those extra permissions to, or write an amendment to increase the number.

There needs to be _some_ limit, 10 seemed reasonable to me but I'm open to making it larger if there's a case.